jdaviz/configs/default/plugins/gaussian_smooth/gaussian_smooth.py

```diff
@@ -88,15 +88,19 @@ def user_api(self):
 
     @observe("dataset_selected", "dataset_items", "stddev", "mode_selected")
     def _set_default_results_label(self, event={}):
-        label_comps = []
-        if hasattr(self, 'dataset') and (len(self.dataset.labels) > 1 or self.app.config == 'mosviz'):  # noqa
-            label_comps += [self.dataset_selected]
-        if self.config == "cubeviz":
-            label_comps += [f"{self.mode_selected.lower()}-smooth"]
+        '''Generate a label and set the results field to that value'''
+        if (hasattr(self, 'dataset') and (len(self.dataset.labels) >= 1) or self.app.config == 'mosviz'):  # noqa
+            dataset = self.dataset_selected
         else:
-            label_comps += ["smooth"]
-        label_comps += [f"stddev-{self.stddev}"]
-        self.results_label_default = " ".join(label_comps)
+            dataset = ''
+
+        smooth_type = (f"{self.mode_selected.lower()}-smooth" if self.config == "cubeviz"
+                       else "smooth")
+        stddev = f"stddev-{self.stddev}"
+
+        # Overriding is allowed, so do not check for uniqueness
+        self.results_label_default = (
+            self.app.return_data_label(f"{dataset} {smooth_type} {stddev}", check_unique=False))
```

Comment:
what benefit does passing this through `return_data_label` provide?

Reply:
In this Gaussian Smooth case, not much. This is because Gaussian Smooth allows the user to overwrite the result. By putting `check_unique=False`, it effectively just returns the string.

The main reason I decided to pass it through anyways is mainly for consistency. Though this is the case with Gaussian Smooth, it may not be the case with other plugins. We should probably centralize our data label generation, especially if we want automatic uniqueness checks. So I pass it through because (1) I don't think it will cause any harm and (2) it can serve as a reminder for future development to follow this paradigm.

I can also just set it if we don't want to

Reply:
@javerbukh was the `return_data_label` intended to be used for plugin labels?  If so, I'd agree it makes sense to start introducing this, but otherwise I'm not sure its worth the overhead since plugin labels have their own logic to check for overwriting, etc.

Reply:
For more context, I originally thought I could provide the smoothing details (e.g. `spatial-smoothed stddev 1.0`) as the `ext` argument, but because we allow overwriting (i.e. don't enforce uniqueness), we can't; `ext` is only used if uniqueness is required

Reply:
The first pass implementation did not include plugin labels because I believe there is already a mixin in utils that handles that (AddResult maybe?). I agree with @duytnguyendtn that I do not see the harm in doing it this way, especially since we may want to centralize all data label generation in the future, but I also see @kecnry point about unnecessary overhead (though if we're not checking for uniqueness it should always be a fast operation). So I don't think I can weigh in too heavily on one side or the other here.